### DISCUSSION TITLE
Facing Error P1000 when trying to connect to a Azure SQL database

### DISCUSSION THREAD

**Discussion by Elucidator2197** [prisma/prisma#24724] [Q&A]
### Question

So first i tried
DATABASE_URL='sqlserver://serverName.database.windows.net:1433;database=databaseNameintegratedSecurity=true;username={[email@homeoffice.com-pany.com](mailto:email@homeoffice.com-pany.com)};password={Password@1};trustServerCertificate=true;'
it gave me an error
Error: Cannot open server "homeoffice.com-pany.com" requested by the login. The login failed.

then i tried
DATABASE_URL='sqlserver://serverName.database.windows.net:1433;database=databaseName;integratedSecurity=true;username={email@[homeoffice.com-pany.com@serverName](mailto:homeoffice.com-pany.com@serverName));password={Password@1};trustServerCertificate=true;'

it gave me an error Error: P1000
Authentication failed against database server

Any suggestions ?
because i know credentials are correct

### Information about Prisma Schema, Client Queries and Environment (optional)

```prisma
// Add your schema.prisma
```

```ts
// Add any relevant Prisma Client queries here
```

- OS: 
- Database: 
- Node.js version: 

- Run `prisma -v` to see your Prisma version and paste it

**Comment by ludralph** [comment]
Hi @Elucidator2197 👋 

Can you please make sure that the username and password are correct and do not contain any special characters that might need to be [URL-encoded](https://www.prisma.io/docs/orm/overview/databases/sql-server#connection-details).  You need to check that you do not have a `$` symbol or other special characters in your password that might need to be URL-encoded.

**Comment by francemazzi** [comment]
Hi @Elucidator2197! It looks like you’re encountering authentication issues with your SQL Server database connection. Based on the errors you're seeing, here are a few suggestions to help resolve them:

Check the Connection String Format:
Ensure that your connection string is correctly formatted. For SQL Server, the format should generally be:

`sqlserver://username:password@serverName.database.windows.net:1433;database=databaseName;trustServerCertificate=true;`
Make sure to replace username, password, serverName, and databaseName with your actual credentials.

Integrated Security:
If you are using integrated security (Windows Authentication), you should not include username and password in the connection string. Integrated security typically looks like this:


`sqlserver://serverName.database.windows.net:1433;database=databaseName;integratedSecurity=true;trustServerCertificate=true;`
If you need to use SQL Server authentication, remove integratedSecurity=true and include username and password.

Check Credentials:
Double-check that your credentials (username and password) are correct and that they have the necessary permissions to access the database.

Server and Port:
Verify that serverName is correctly specified and that it includes the appropriate domain. Ensure that port 1433 is open and accessible from your environment.

Firewall and Network Settings:
Ensure that your database server allows connections from your IP address and that there are no firewall rules blocking access.

Trust Server Certificate:
If you are using trustServerCertificate=true, ensure that this setting is supported and correctly configured in your environment.

Testing Connection:
Try connecting to the database using another tool (like SQL Server Management Studio) to verify that the credentials and server details are correct.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
